data/processing/input_latency.html

```diff
@@ -0,0 +1,319 @@
+        queue_uninteresting.push({"title":"rendererIdle", "duration":queue_renderer_idle});
+        handle_uninteresting.push({"title":"rendererIdle", "duration":handle_renderer_idle});
+        queue_active_uninteresting += queue_renderer_idle;
+        handle_active_uninteresting += handle_renderer_idle; 
+
+        // Calculate uninteresting time, which is all unaccounted for time (total time minus the 
+        // active time, which are the events we consider interesting plus the time spent idle, what
+        // remains is the time spent in uninteresting tasks) 
+        var queue_uninteresting_time = (latency_mid - latency_start) - queue_active_uninteresting;
+        queue_uninteresting.push({"title":"uninteresting", "duration":queue_uninteresting_time});        
+        queue_active_uninteresting += queue_uninteresting_time;
+        var handle_uninteresting_time = (latency_end - latency_mid) - handle_active_uninteresting;
+        handle_uninteresting.push({"title":"uninteresting", "duration":handle_uninteresting_time});
+        handle_active_uninteresting += handle_uninteresting_time;
+
+	results.push({"Latency Duration":latency_duration, "Queue Events Bottom-Up":mergeSameNameEvents(queue), 
+                      "Handling Events Bottom-Up":mergeSameNameEvents(handle), "Bottom-Up Event Sum":queue_active + handle_active, 
+                      "Queue Events Top-Down":mergeSameNameEvents(queue_uninteresting, "duration"), 
+                      "Handling Events Top-Down":mergeSameNameEvents(handle_uninteresting, "duration"), 
+                      "Top-Down Event Sum":queue_active_uninteresting + handle_active_uninteresting});
+      }
+    }
+    result.addPair("Latencies", results);
+  }
+
+  tr.mre.FunctionRegistry.register(inputLatencyMapper);
+
+  return {
+    inputLatencyMapper,
+  };
+
+  function getInputLatencyEvents(browserHelper) {
+    const isInputLatencyEvent = e => e.title === LATENCY_EVENT;
+    return browserHelper.getAllAsyncSlicesMatching(isInputLatencyEvent);
+  }
+
+  function getRendererEventsInRange(browserHelper, rendererID, range_of_interest) {
+    return browserHelper.getAllAsyncSlicesMatching(function(slice) {
+      return (slice.parentContainer.getProcess().pid === rendererID &&
+              slice.parentContainer.name === "CrRendererMain" && 
+              slice.category != "disabled-by-default-renderer.scheduler" &&
+              slice.category != "renderer.scheduler" &&
+              slice.category != "blink.animations,devtools.timeline,benchmark,rail" &&
+              slice.start >= range_of_interest.min &&
+              slice.end <= range_of_interest.max)
+    });
+  }
+
+  // Given a set of slices, get all the top level slices within time bounds
+  function getTopLevelSlices(slices, start, end) {
+    var top_level = [];
+    for (const slice of slices) {
+      // If the top level slice is not already in the top level slices list
+      if (typeof slice.mostTopLevelSlice !== "undefined"
```

Comment:
These events are on the renderer but not events (like Animation) but I can't filter them out using category like I do with 'renderer.scheduler' because their category is 'toplevel' and there are some 'toplevel' events that I don't want to filter out.